scipy/stats/_stats_py.py

```diff
@@ -5552,6 +5659,200 @@ def _mgc_stat(distx, disty):
+    Let :math:`D^x` be the :math:`n \times n` distance matrix of :math:`x` and :math:`D^y` be the :math:`n \times n` be the 
+    distance matrix of :math:`y`. The distance covariance is
+    
+    .. math::
+
+        Dcov_n^b(x,y) = (1/(n^2)) \ times \mathrm{tr(D^{y} H D^{y} H)}
+
+    where math`\mathrm{tr} (\cdot)` is the trace operator and :math:`H` is defined as \mathbb{1} =I−(1/n)J where :math:`I` 
+    is the identity matrix and :math:`J` is a matrix of ones. The normalized version 
+    of this covariance is distance correlation 1 and is
+    
+    .. math::
+
+        Dcorr_n^b(x,y) = {Dcov_n^b(x,y)}/{\sqrt{Dcov_n^b(x,x) Dcov_n^b(y,y)}}
+
+    This is a biased test statistic. An unbiased alternative also exists, 
+    and is defined using the following: 
+
+    Consider the centering process where \mathbb{1} is the indicator function:
+    
+    .. math::
+
+        C_{ij}^x = [\(D_{ij}^x - (1/(n-2)) \sum_{t=1}^{n} D_{it}^x - (1/(n-2)) 
+                \sum_{s=1}^{n} D_{sj}^x + (\frac{1}{(n-2)(n+2)}) \sum_{s,t=1}^{n} D_{st}^x] 1_{i \neq j}
+    
+    and similarly for :math:`C^y`. 
+
+    Then, the unbiased Dcorr test statistic is:
+    
+    .. math:: 
+        
+        Dcov_n(x,y)=(1/n(n−3))\mathrm{tr(C^x C^y)}
+
+    The p-value returned is calculated using a permutation test using a helper function
+    that calculates the parallel p-value.
+
+    Dcorr's stat and p-value is equivalent to the values in similar testing methods
+    such as Energy and hSIC [1]_ [2]_. 
+
+    References
+    ----------
+    .. [1] Panda, S., Cencheng, S., Perry, R., Zorn, J., Lutz, A.
+           Priebe, C., Vogelstein, J. T. (2019). Nonpar MANOVA via Independence
+           Testing. Package. : arXiv: `1910.08883`
+    .. [2] Cencheng, S., Vogelstein, J. T. (2019). The Exact Equivalence of
+           Distance and Kernel Methods for Hypothesis Testing. 
+           Package. : arXiv: `1806.05514`
+
+    Examples
+    --------
+    >>> from scipy.stats import distance_correlation
+    >>> x = np.arange(100)
+    >>> y = x
+    >>> stat, pvalue = distance_correlation(x, y, workers=-1)
+    >>> '%.1f, %.3f' % (stat, pvalue)
+    '1.0, 0.001'
+
+    To run an unpaired two-sample test,
+
+    >>> x = np.arange(100)
+    >>> y = np.arange(79)
+    >>> stat, pvalue = distance_correlation(x, y, random_state=1)
+    >>> '%.3f, %.2f' % (stat, pvalue)
+    '0.033, 0.02'
+
+    or, if shape of the inputs are the same,
+    
+    >>> x = np.arange(100)
+    >>> y = x
+    >>> stat, pvalue = distance_correlation(x, y, is_twosamp=True)
+    >>> '%.3f, %.1f' % (stat, pvalue)
+    '-0.008, 1.0'
+    """
+    x, y = _check_inputs(x,y, compute_distance=compute_distance, reps=reps,
+                         workers=workers, is_twosamp=is_twosamp, random_state=random_state)
+
+    # calculate Distance_Correlation stat
+    stat = _dcorr(x,y)
+
+    # calculate permutation p-value
+    pvalue, null_dist = _perm_test(x, y, stat, func=_dcorr, reps=reps, workers=workers,
```

Comment:
change to `calc_stat`

Reply:
https://github.com/kfangster/scipy/pull/1/commits/b36584b9389aca5864c346516c655032c589bee0